### ISSUE TITLE
Suspected parse error for $this->_compile_{$section}

### ISSUE THREAD

**Issue by nevadascout** [HvyIndustries/crane#150]
Within this class constructor the line `$this->_compile_{$section} = TRUE;` doesn't appear to get parsed properly.

``` php
public function __construct($config = array())
{
    $this->CI =& get_instance();
    $this->CI->load->language('profiler');

    // default all sections to display 
    foreach ($this->_available_sections as $section)
    {
        if ( ! isset($config[$section]))
        {
            $this->_compile_{$section} = TRUE;
        }
    }

    $this->set_sections($config);
    log_message('info', 'Profiler Class Initialized');
}
```

Nothing is parsed past that line, and it sets some array items to `undefined` which breaks things further down the line.

_Framework: **CodeIgniter 3.1.0**_

**Comment by nevadascout** [comment]
@ichiriac any ideas on this? It's definitely the `{$section}` part of the line `$this->_compile_{$section}` that is breaking.

It seems to not parse the file any further than that line

**Comment by RazaLabs** [comment]
@nevadascout any update on this?

**Comment by ichiriac** [comment]
@nevadascout this bug is actually fixed on the current master version, if you don't use lines and columns properties from the lexer but instead offsets you could give a try on the master, it contains many others bugfixes. 

I will finish the full release in few days, remains some little word to be done.

**Comment by ichiriac** [comment]
done

**Comment by RazaLabs** [comment]
@ichiriac kindly can you please tell me how can I make it work with codeigniter?
Do I have to get instance with? $this->CI =& get_instance();
or add any json/js file in root?

**Comment by ichiriac** [comment]
@RazaLabs the easy way is just to wait @nevadascout to make a new release, including this new version of php-parser covered with tests and maybe some adaptation.

The hard way is to locate and modify this file : 
https://github.com/HvyIndustries/crane/blob/master/client/package.json#L126

Instead put : `"php-parser": "^0.1.0",`, and next from the cli run `npm update`, then in principle by restarting VS Code it should be OK (but there is some changes in AST Tree on use statements, so the new version may not be fully compliant with actual code). 

> It's not garantee to work so make a copy of the package.json and an re-run `npm update` to rollback if it fails

**Comment by TheColorRed** [comment]
@ichiriac I have tried integrating the new parser into a new build, and when I do it doesn't seem to parse the files... I am not sure why. When I rollback the parser to `0.0.10` file parsing works again. Here is what we have:

```ts
            var parserInst = phpParser.create({
                parser: {
                    locations: true,
                    docBlocks: true,
                    suppressErrors: true
                }
            });

            var ast = parserInst.parseCode(text);
```

**Comment by ichiriac** [comment]
Do you get an error message (like undefined offsets) ?

You may have to adapt this code :
https://github.com/HvyIndustries/crane/blob/master/server/src/hvy/treeBuilder.ts#L142

See explanations here :
https://github.com/glayzzle/php-parser/issues/19#issuecomment-257143574

**Comment by TheColorRed** [comment]
It looks like it is somehow disrupting the language client, as it doesn't even get to the point of processing the file(s).

```ts
console.log('sending request to langClient'); // This line works
Crane.langClient.sendRequest({ method: "buildFromFiles" }, {/* information values */ }); // This line should be working
```

In the server when the above request is sent we then have a `console.log` as the first thing:
```ts
connection.onRequest(buildFromFiles, (project) => {
    connection.console.log('building from files'); // This doesn't get printed
    /*process the files here*/
});
```

Using the old parser both `sending request to langClient` and `building from files` line is displayed in the console. 
Using the new one, only `sending request to langClient` is displayed in the console.

Sorry that I don't have more information, but I am not getting any error messages or anything...

**Comment by TheColorRed** [comment]
Not sure why it can't find the lexer...

If I move the `require` into the `Promise` on `TreeBuilder.Parser()`, I am able to get an error:

```
{ Error: ENOENT: no such file or directory, scandir 'c:\Users\rnaddy\Documents\vscode\projects\crane\client\server\node_modules\php-parser\src\lexer'
    at Error (native)
    at Object.fs.readdirSync (fs.js:951:18)
    at Object.fs.readdirSync (ELECTRON_ASAR.js:571:28)
    at module.exports (c:\Users\rnaddy\Documents\vscode\projects\crane\client\server\node_modules\php-parser\src\lexer.js:205:6)
    at Object.<anonymous> (c:\Users\rnaddy\Documents\vscode\projects\crane\client\server\node_modules\php-parser\main.js:111:38)
    at Module._compile (module.js:556:32)
    at Object.Module._extensions..js (module.js:565:10)
    at Module.load (module.js:473:32)
    at tryModuleLoad (module.js:432:12)
    at Function.Module._load (module.js:424:3)
  errno: -4058,
  code: 'ENOENT',
  syscall: 'scandir',
  path: 'c:\\Users\\rnaddy\\Documents\\vscode\\projects\\crane\\client\\server\\node_modules\\php-parser\\src\\lexer' }
```

**Comment by ichiriac** [comment]
oups ^^ https://github.com/glayzzle/php-parser/blob/master/.npmignore#L3, it's because the old lexer syntax was not needed to run, i will make a new release to fix it

**Comment by ichiriac** [comment]
it's fixed now on 0.1.1

**Comment by TheColorRed** [comment]
That seems to have fixed the issue, and it is parsing files again!

**Comment by TheColorRed** [comment]
WOW that ran fast!

**Comment by ichiriac** [comment]
FYI, actually I've only found 2 edge cases reported here https://github.com/glayzzle/php-parser/issues/21 and https://github.com/glayzzle/php-parser/issues/20 - in principle they should not break the parsing in silent mode.

**Comment by TheColorRed** [comment]
@ichiriac The only Issue that I am getting is this one:

```
TypeError: Cannot read property '0' of undefined
    at Object.parse (c:\Users\rnaddy\Documents\vscode\projects\crane\client\server\node_modules\php-parser\src\parser.js:190:26)
    at Object.parseCode (c:\Users\rnaddy\Documents\vscode\projects\crane\client\server\node_modules\php-parser\main.js:78:19)
    at c:\Users\rnaddy\Documents\vscode\projects\crane\client\server\hvy\treeBuilder.js:40:34
    at TreeBuilder.Parse (c:\Users\rnaddy\Documents\vscode\projects\crane\client\server\hvy\treeBuilder.js:32:16)
    at c:\Users\rnaddy\Documents\vscode\projects\crane\client\server\server.js:211:25
    at c:\Users\rnaddy\Documents\vscode\projects\crane\client\server\node_modules\filequeue\lib\filequeue.js:77:22
    at tryToString (fs.js:455:3)
    at FSReqWrap.readFileAfterClose [as oncomplete] (fs.js:442:12)
```

The rest are in the outdated treebuilder

**Comment by nevadascout** [comment]
@TheColorRed are you updating the treebuilder, or do you want me to do it? (Let's avoid both doing it!)

**Comment by TheColorRed** [comment]
If you have time, you can do it. If not I can take a look in the next day or two.

**Comment by nevadascout** [comment]
Aright, leave it to me. I want to get this updated and released by the end of this week.

**Comment by nevadascout** [comment]
Released in v0.3.0
